tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,155 @@
+        """
+        We mock request.Request method to raise a `Timeout` and expect the tap to retry this up to 
+        5 times when empty string value of `request_timeout` passed
+        """
+        config = {'token': '123', "request_timeout": ""}
+        client = client_.Client(config)
+        url = client.build_url(self.endpoint)
+        try:
+            client.request('GET', url)
+        except Timeout as e:
+            pass
+
+        # Verify that request.Request called 5 times
+        self.assertEqual(mocked_request.call_count, 5)
+        
+    @patch('requests.Request', side_effect=Timeout)
+    def test_request_timeout_backoff_for_zero_string_value_in_config(self, mocked_request, mock_sleep):
+        """
+        We mock request.Request method to raise a `Timeout` and expect the tap to retry this up to 
+        5 times when zero string value of `request_timeout` passed
+        """
+        config = {'token': '123', "request_timeout": "0"}
+        client = client_.Client(config)
+        url = client.build_url(self.endpoint)
+        try:
+            client.request('GET', url)
+        except Timeout as e:
+            pass
+
+        # Verify that request.Request called 5 times
+        self.assertEqual(mocked_request.call_count, 5)
+        
+    @patch('requests.Request', side_effect=Timeout)
+    def test_request_timeout_backoff_for_zero_int_value_in_config(self, mocked_request, mock_sleep):
+        """
+        We mock request.Request method to raise a `Timeout` and expect the tap to retry this up to 
+        5 times when int 0 value of `request_timeout` passed
+        """        
+        config = {'token': '123', "request_timeout": 0}
+        client = client_.Client(config)
+        url = client.build_url(self.endpoint)
+        try:
+            client.request('GET', url)
+        except Timeout as e:
+            pass
+
+        # Verify that request.Request called 5 times
+        self.assertEqual(mocked_request.call_count, 5)
+        
+    @patch('requests.Request', side_effect=Timeout)
+    def test_request_timeout_backoff_for_none_value_in_config(self, mocked_request, mock_sleep):
+        """
+        We mock request.Request method to raise a `Timeout` and expect the tap to retry this up to 
+        5 times when `request_timeout` does not passed
+        """        
+        config = {'token': '123', "request_timeout": 0}
+        client = client_.Client(config)
+        url = client.build_url(self.endpoint)
+        try:
+            client.request('GET', url)
+        except Timeout as e:
+            pass
+
+        # Verify that request.Request called 5 times
+        self.assertEqual(mocked_request.call_count, 5)
+
+@patch('time.sleep')
+class TestConnectionError(unittest.TestCase):
+    
+    endpoint = "forms"
+
+    @patch('requests.Request', side_effect=Timeout)
+    def test_connection_error_backoff(self, mocked_request, ConnectionError):
```

Comment:
Update above test case as it's for `ConnectionError` but you mocked the request to raise `Timeout`. Also, you provided `ConnectionError` as a mocked argument of `time.sleep`.

Reply:
Updated.